omnigibson/object_states/toggle.py

```diff
@@ -51,16 +52,21 @@ def _initialize(self):
         pre_existing_mesh = get_prim_at_path(mesh_prim_path)
         # Create a primitive mesh if it doesn't already exist
         if not pre_existing_mesh:
-            button = UsdGeom.Sphere.Define(og.sim.stage, Sdf.Path(mesh_prim_path))
             self.radius = m.DEFAULT_RADIUS if self.radius is None else self.radius
+            mesh = create_primitive_mesh(
+                prim_path=mesh_prim_path,
+                primitive_type="Sphere",
+                extents=1.0,
+            )
         else:
             # Infer radius from mesh if not specified as an input
-            self.radius = pre_existing_mesh.GetAttribute("radius").Get() if self.radius is None else self.radius
+            recompute_extents(prim=pre_existing_mesh)
+            scale = np.array(pre_existing_mesh.GetAttribute("xformOp:scale").Get())
```

Comment:
This is assuming the preexisting meta mesh is sphere mesh, not sphere shape, right?

Have you made the change in ig_pipeline code and re-export the dataset?

Reply:
Yes, I have made the change in the ig_pipeline code. Waiting for Cem to re-export to verify that everything works as expected.

Reply:
Got it.